lib/rex/parser/arguments.rb

```diff
@@ -12,21 +12,42 @@ module Parser
     #
     ###
     class Arguments
+      # e.g. '-x' or '-xyz'
+      SHORT_FLAG = /^-[a-zA-Z]+$/.freeze
+      # e.g. '--verbose', '--very-verbose' or '--very-very-verbose'
+      LONG_FLAG = /^--([a-zA-Z]+)((-[a-zA-Z]+)*)$/.freeze
+
       #
       # Initializes the format list with an array of formats like:
       #
       # Arguments.new(
       #    '-b'                => [ false, "some text"                 ],
       #    ['-b']              => [ false, "some text"                 ],
-      #    '--sample'          => [ false, "sample long arg"           ],
-      #    '--also-a-sample'   => [ false, "sample longer arg"         ],
-      #    ['-x', '--execute'] => [ true, "mixing long and short args" ]
+      #    ['-x', '--execute'] => [ true, "mixing long and short args" ],
+      #    ['-t', '--test']    => [ true, "testing custom <opt> value", "<arg_to_test>" ]
       # )
       #
       def initialize(fmt)
-        normalised_fmt = fmt.map { |key, metadata| [Array(key), metadata] }.to_h
+        normalised_fmt = fmt.map do |key, metadata|
+          if metadata[0] && !metadata[2]
+            metadata = metadata.clone
+            metadata.push('<opt>')
+          end
+          [Array(key), metadata]
+        end.to_h
+
+        # Runtime check to see if it has been initialised correctly.
+        # We need to have a short arg as that is what we return
+        normalised_fmt.each_key do |key_array|
+          includes_short_flag = false
+          key_array.each { |key| includes_short_flag = true if key =~ SHORT_FLAG }
+          unless includes_short_flag
+            raise "Failed to create a Rex::Parser::Arguments object. No short-hand arguments given for #{key_array}."
```

Comment:
Not strongly opinionated; I'm not sure if a short flag should always be required 👀

Reply:
I've decided to yield the short flag is possible, and then defaulting to returning the long flag if that exists.